pyotgw/protocol.py

```diff
@@ -62,9 +63,17 @@ def connection_lost(self, exc):
         self.transport.close()
         for q in [self._cmdq, self._updateq]:
             while not q.empty():
-                q.get()
-        if self._watchdog_task is not None:
-            self._watchdog_task.cancel()
+                try:
+                    q.get_nowait()
+                except Empty:
+                    continue
+
+        # If you cancel the watchdog task here, it never
+        # gets triggerd, and thus never reconnects. Commented for now.
+
+        # if self._watchdog_task is not None:
+        # self._watchdog_task.cancel()
+
```

Comment:
I noticed the same in my preliminary work, comments may be removed here.

Reply:
Resolved in ac3c3ab